src/deluge/model/instrument/midi_instrument.h

```diff
@@ -45,6 +45,14 @@ class MIDIInstrument final : public NonAudioInstrument {
 	bool readTagFromFile(Deserializer& reader, char const* tagName);
 	Error readModKnobAssignmentsFromFile(int32_t readAutomationUpToPos,
 	                                     ParamManagerForTimeline* paramManager = nullptr);
+
+	// cc names
+	String ccNames[kNumRealCCNumbers];
```

Comment:
Couple things - 

1. This is a 600 byte array, it would be better to store it externally instead of as part of the output
2. If you make it a std array then it can use iterators instead of a for loop to check all the names. Then this can change to a pointer to an array, and you can allocate the actual array in the midi output constructor
3. This should be private to enforce access via the api functions you wrote

Reply:
> Couple things -
> 
> 1. This is a 600 byte array, it would be better to store it externally instead of as part of the output
> 2. If you make it a std array then it can use iterators instead of a for loop to check all the names. Then this can change to a pointer to an array, and you can allocate the actual array in the midi output constructor
> 3. This should be private to enforce access via the api functions you wrote

Will do thanks!